### DISCUSSION TITLE
Caching Jekyll _site files not working

### DISCUSSION THREAD

**Discussion by datapolitical** [community/community#26816] [Actions]
<p>I’m trying to cache my _site files when I build my Jekyll site using this:</p>
<pre><code>  # Use GitHub Actions' cache to cache site on servers
  - uses: actions/cache@v2
    with:
      path: _site
      key: ${{ runner.os }}-site-${{ hashFiles('**/index.html') }}
      restore-keys: |
        ${{ runner.os }}-site-
</code></pre>
<aside class="onebox githubblob">
  <header class="source">
      <a href="https://github.com/datapolitical/chrisfnicholson.github.io/blob/1be6c49ebfa3861869820e4f2ef18f4a449b48b2/.github/workflows/build-jekyll.yml" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>
  <article class="onebox-body">
    <h4><a href="https://github.com/datapolitical/chrisfnicholson.github.io/blob/1be6c49ebfa3861869820e4f2ef18f4a449b48b2/.github/workflows/build-jekyll.yml" target="_blank" rel="noopener nofollow ugc">datapolitical/chrisfnicholson.github.io/blob/1be6c49ebfa3861869820e4f2ef18f4a449b48b2/.github/workflows/build-jekyll.yml</a></h4>
<pre><code class="lang-yml">name: Build and Deploy to Github Pages

on:
  push:
    branches:
      - main  # Here source code branch is `master`, it could be other branch
  workflow_dispatch:

jobs:
  build_and_deploy:
    runs-on: ubuntu-20.04
    steps:
      - name: 'Wait for status checks'
        id: waitforstatuschecks
        uses: "WyriHaximus/github-action-wait-for-status@v1.2"
        with:
          ignoreActions: build_and_deploy
          checkInterval: 10
        env:
          GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
</code></pre>

  This file has been truncated. <a href="https://github.com/datapolitical/chrisfnicholson.github.io/blob/1be6c49ebfa3861869820e4f2ef18f4a449b48b2/.github/workflows/build-jekyll.yml" target="_blank" rel="noopener nofollow ugc">show original</a>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>Unfortunately when it runs I get:</p>
<p>Received 22 of 22 (100.0%), 0.0 MBs/sec<br>
Cache Size: ~0 MB (22 B).</p>
<p>Not clear what’s going wrong. Is the path incorrect?</p>

**Comment by Simran-B** [comment, marked as answer]
<p>I don’t think that Jekyll can utilize what it previously generated if you cache the <code>_site</code> folder. You should cache the folder <code>.jekyll-cache</code> and if it exists also the file <code>.jekyll-metadata</code> (haven’t seen it on Windows, but on Linux). Maybe cache the <code>_site</code> in addition, not sure if it makes a difference.</p>

**Comment by datapolitical** [comment]
<p>Thanks! Yea was able to eventually get this working and it’s much better!</p>
